addons/binding/org.openhab.binding.squeezebox/src/main/java/org/openhab/binding/squeezebox/handler/SqueezeBoxServerHandler.java

```diff
@@ -0,0 +1,825 @@
+		sendCommand(mac + " show line1:" + line + " duration:"
+				+ String.valueOf(duration));
+	}
+
+	public void showStringHuge(String mac, String line) {
+		showStringHuge(mac, line, 5);
+	}
+
+	public void showStringHuge(String mac, String line, int duration) {
+		sendCommand(mac + " show line1:" + line + " font:huge duration:"
+				+ String.valueOf(duration));
+	}
+
+	public void showStrings(String mac, String line1, String line2) {
+		showStrings(mac, line1, line2, 5);
+	}
+
+	public void showStrings(String mac, String line1, String line2, int duration) {
+		sendCommand(mac + " show line1:" + line1 + " line2:" + line2
+				+ " duration:" + String.valueOf(duration));
+	}
+
+	/**
+	 * Send a generic command to a given player
+	 * 
+	 * @param playerId
+	 * @param command
+	 */
+	public void playerCommand(String mac, String command) {
+		sendCommand(mac + " " + command);
+	}
+
+	/**
+	 * Ask for player list
+	 */
+	public void requestPlayers() {
+		sendCommand("players 0");
+	}
+
+	/**
+	 * Send a command to the Squeeze Server.
+	 */
+	private synchronized void sendCommand(String command) {
+		if (!isConnected()) {
+			logger.debug("No connection to SqueezeServer, will attempt to reconnect now...");
+			connect();
+			if (!isConnected()) {
+				logger.error(
+						"Failed to reconnect to SqueezeServer, unable to send command {}",
+						command);
+				return;
+			}
+		}
+		logger.debug("Sending command: {}", command);
+		try {
+			BufferedWriter writer = new BufferedWriter(new OutputStreamWriter(
+					clientSocket.getOutputStream()));
+			writer.write(command + NEW_LINE);
+			writer.flush();
+		} catch (IOException e) {
+			logger.error("Error while sending command to Squeeze Server ("
+					+ command + ")", e);
+		}
+	}
+
+	/**
+	 * Connects to a SqueezeBox Server
+	 */
+	private void connect() {
+		SqueezeBoxServerConfig config = getConfigAs(SqueezeBoxServerConfig.class);
+		this.host = config.ipAddress;
+		this.cliport = config.cliport;
+		this.webport = config.webport;
+
+		try {
+			clientSocket = new Socket(host, cliport);
+		} catch (IOException e) {
+			logger.error("Failed to connect to the Squeeze Server at "
+					+ config.ipAddress + ":" + config.cliport, e);
+			return;
+		}
+
+		try {
+			listener = new SqueezeServerListener();
+			listener.start();
+		} catch (IllegalThreadStateException e) {
+			logger.error("Failed to start the Squeeze Server listener thread",
+					e);
+			return;
+		}
+
+		updateStatus(ThingStatus.ONLINE);
+		logger.info("Squeeze Server connection started.");
+	}
+
+	/**
+	 * Disconnects from a SqueezeBox Server
+	 */
+	private void disconnect() {
+		// if (!isConnected())
+		// return;
+
+		try {
+			listener.terminate();
+			clientSocket.close();
+		} catch (IOException e) {
+			logger.error("Error attempting to disconnect from Squeeze Server",
+					e);
+			return;
+		} finally {
+			clientSocket = null;
+			listener = null;
+		}
+		players.clear();
+		updateStatus(ThingStatus.OFFLINE);
```

Comment:
pls remove this line - in the dispose() method, it is not possible to set the Thing status anymore.